packages/yew/src/functional/mod.rs

```diff
@@ -155,18 +215,11 @@ where
     }
 
     fn destroy(&mut self, _ctx: &Context<Self>) {
-        let mut hook_state = self.hook_state.borrow_mut();
-        for hook in hook_state.destroy_listeners.drain(..) {
-            hook()
-        }
-    }
-}
+        let mut hook_ctx = self.hook_ctx.borrow_mut();
 
-pub(crate) fn get_current_scope() -> Option<AnyScope> {
-    if CURRENT_HOOK.is_set() {
-        Some(CURRENT_HOOK.with(|state| state.scope.clone()))
-    } else {
-        None
+        for hook in hook_ctx.hooks.drain(..) {
+            drop(hook);
+        }
```

Comment:
```suggestion
        drop(std::mem::take(&mut hook_ctx.hooks));
```

Reply:
This is to make sure that the destructor is called in the same order as the position in the hook state.

Reply:
I see, I didn't realize that vec doesn't guarantee a specific drop order, all good then.